api/controllers/blocks.js

```diff
@@ -39,7 +40,49 @@ router.get(`${endpoint}/:blockId`, async (req, res) => {
       throw `That block does not belong to ${username}`;
     }
   } catch (error) {
-    res.status(404).send(error);
+    res.status(401).send(error);
+  }
+});
+
+// update
+router.put(`${endpoint}/:blockId`, async (req, res) => {
+  const blockId = parseInt(req.params.blockId);
+  const username = req.user.username;
+
+  try {
+    // check to make sure current user has this id assigned to them
+    const userHasBlock = await users.hasBlock(username, blockId);
+
+    if (userHasBlock) {
+      const block = await blocks.update(blockId, req.body);
+      res.status(200).json(block);
+    } else {
+      throw `That block does not belong to ${username}`;
+    }
+  } catch (error) {
+    res.status(401).send(error);
```

Comment:
i think a 404 might be a more semantic status code for not being able to find the block.  401 should really only be used for failed authentication (in the auth middlware).